src/stream_map.rs

```diff
@@ -113,6 +114,36 @@ where
             Some((id, None)) => Poll::Ready((id, None)),
         }
     }
+
+    /// Returns an iterator over all streams whose inner type is `T`.
```

Comment:
I think from the perspective of the user, there is no such thing as an "inner" type. `T` will be the same type as to what they passed in `try_push`. The fact that we are wrapping that is an implementation detail.

Reply:
We should probably mention this in the docs.

Reply:
See https://github.com/thomaseizinger/rust-futures-bounded/pull/10#discussion_r2077445109.

Reply:
```suggestion
    /// Returns an iterator over all streams of type `T` pushed via [`StreamMap::try_push`].
```

Something like this?

Reply:
Done